shoplist/build.gradle

```
@@ -0,0 +1,59 @@
+plugins {
+    id 'com.android.library'
+    id 'kotlin-android'
+}
+
+android {
+    compileSdkVersion 30
+    buildToolsVersion "30.0.3"
+
+    defaultConfig {
+        minSdkVersion 21
+        targetSdkVersion 30
+        versionCode 1
+        versionName "1.0"
+
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+        consumerProguardFiles "consumer-rules.pro"
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+    }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
+    buildFeatures.viewBinding true
+}
+
+dependencies {
+
+    api "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
+    api 'androidx.core:core-ktx:1.3.2'
+    api 'androidx.appcompat:appcompat:1.2.0'
+    api 'com.google.android.material:material:1.3.0'
+    api 'androidx.lifecycle:lifecycle-livedata-ktx:2.3.1'
+    api 'androidx.lifecycle:lifecycle-viewmodel-ktx:2.3.1'
+    api 'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.4.3'
+    api 'io.coil-kt:coil:1.2.1'
+
+    api "com.google.dagger:dagger-android:2.33"
+    api "com.google.dagger:dagger-android-support:2.33"
+    api 'com.squareup.okhttp3:okhttp:4.9.1'
+    api 'com.squareup.okhttp3:logging-interceptor:4.9.1'
+    api 'com.squareup.retrofit2:retrofit:2.9.0'
+    api 'com.jakewharton.retrofit:retrofit2-kotlinx-serialization-converter:0.8.0'
+    api 'org.jetbrains.kotlinx:kotlinx-serialization-core:1.1.0'
+    api 'org.jetbrains.kotlinx:kotlinx-serialization-json:1.1.0'
```

There's no need to use api here,
you can replace them with implementation. 

```suggestion
    implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
    implementation 'androidx.core:core-ktx:1.3.2'
    implementation 'androidx.appcompat:appcompat:1.2.0'
    implementation 'com.google.android.material:material:1.3.0'
    implementation 'androidx.lifecycle:lifecycle-livedata-ktx:2.3.1'
    implementation 'androidx.lifecycle:lifecycle-viewmodel-ktx:2.3.1'
    implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.4.3'
    implementation 'io.coil-kt:coil:1.2.1'

    implementation "com.google.dagger:dagger-android:2.33"
    implementation "com.google.dagger:dagger-android-support:2.33"
    implementation 'com.squareup.okhttp3:okhttp:4.9.1'
    implementation 'com.squareup.okhttp3:logging-interceptor:4.9.1'
    implementation 'com.squareup.retrofit2:retrofit:2.9.0'
    implementation 'com.jakewharton.retrofit:retrofit2-kotlinx-serialization-converter:0.8.0'
    implementation 'org.jetbrains.kotlinx:kotlinx-serialization-core:1.1.0'
    implementation 'org.jetbrains.kotlinx:kotlinx-serialization-json:1.1.0'
```

---

Note that, many of the dependencies are outdated.
we can use a newer version 😉